src/PNetSFInterface.cc

```diff
@@ -0,0 +1,423 @@
+
+        // 300 <= pT < 400
+        // HP and MP SFs not yet available - default to 1
+        scaleFactorsDY_medPt_[6] = (1.10929824);                 // LP central
+        scaleFactorsDY_medPt_[7] = (1.10929824 + ( 0.31057888)); // LP up
+        scaleFactorsDY_medPt_[8] = (1.10929824 + (-0.31057888)); // LP down
+
+        // pT >= 400
+        scaleFactorsDY_highPt_[6] = (0.65913237);                 // LP central
+        scaleFactorsDY_highPt_[7] = (0.65913237 + ( 0.27456928)); // LP up
+        scaleFactorsDY_highPt_[8] = (0.65913237 + (-0.27456928)); // LP down
+    }
+    else if (period == "2017")
+    {
+        // pT < 300
+        // HP and MP SFs not yet available - default to 1
+        scaleFactorsDY_lowPt_[6] = (1.00916880);                // LP central
+        scaleFactorsDY_lowPt_[7] = (1.00916880 + ( 0.1431627)); // LP up
+        scaleFactorsDY_lowPt_[8] = (1.00916880 + (-0.1431627)); // LP down
+
+        // 300 <= pT < 400
+        // HP and MP SFs not yet available - default to 1
+        scaleFactorsDY_medPt_[6] = (1.38009424);                 // LP central
+        scaleFactorsDY_medPt_[7] = (1.38009424 + ( 0.24822319)); // LP up
+        scaleFactorsDY_medPt_[8] = (1.38009424 + (-0.24822319)); // LP down
+
+        // pT >= 400
+        scaleFactorsDY_highPt_[6] = (1.08742263);                 // LP central
+        scaleFactorsDY_highPt_[7] = (1.08742263 + ( 0.27418802)); // LP up
+        scaleFactorsDY_highPt_[8] = (1.08742263 + (-0.27418802)); // LP down
+    }
+    else if (period == "2018" || period == "2022" || period == "2023")
+    {
+        // pT < 300
+        // HP and MP SFs not yet available - default to 1
+        scaleFactorsDY_lowPt_[6] = (1.17889197);                // LP central
+        scaleFactorsDY_lowPt_[7] = (1.17889197 + ( 0.1441964)); // LP up
+        scaleFactorsDY_lowPt_[8] = (1.17889197 + (-0.1441964)); // LP down
+
+        // 300 <= pT < 400
+        // HP and MP SFs not yet available - default to 1
+        scaleFactorsDY_medPt_[6] = (1.09308159);                 // LP central
+        scaleFactorsDY_medPt_[7] = (1.09308159 + ( 0.18257319)); // LP up
+        scaleFactorsDY_medPt_[8] = (1.09308159 + (-0.18257319)); // LP down
+
+        // pT >= 400
+        scaleFactorsDY_highPt_[6] = (1.24201816);                 // LP central
+        scaleFactorsDY_highPt_[7] = (1.24201816 + ( 0.27224163)); // LP up
+        scaleFactorsDY_highPt_[8] = (1.24201816 + (-0.27224163)); // LP down
+    }
+    else
+    {
+        std::string errorMessage = "PNseSFInterface - Invalid period: "
+                                 + period
+                                 + " [options are: 2016preVFP/2016postVFP/2017/2018/2022/2023]";
+        throw std::logic_error(errorMessage);
+    }
+}
+
+// Get the SF vector based on jet pT and sampleType
+std::vector<float> PNetSFInterface::getSFvec(const float pT, const std::string sampleType)
+{
```

Comment:
Can you include in a comment here a reference for the pT edges you consider to distinguish between low, medium and high pT? Maybe one of your presentations?